hyperspy/_signals/signal1d.py

```diff
@@ -1045,20 +1059,24 @@ def remove_background(
             background_type='Power Law',
             polynomial_order=2,
             fast=True,
+            module=None,
             show_progressbar=None, display=True, toolkit=None):
-        signal_range = signal_range_from_roi(signal_range)
+
+        if module is None:
```

Comment:
What about moving the auto numexpr code to the individual components written with ``Expression``. Ideally we could move it to ``Expression`` itself but ``numexpr`` does not support as many functions as numpy.

In this way:
* The speed improvement would be available everywhere where the components are used.
* We can reduce the number of kwargs of ``remove_background``.

Reply:
By the way, we could add numexpr to the requirements: it just depends on numpy and is pypi and conda installable.